pkg/statemetrics/mr_state_metrics.go

```diff
@@ -104,11 +104,8 @@ func (r *MRStateRecorder) Record(ctx context.Context, mrList resource.ManagedLis
 	}
 
 	mrs := mrList.GetItems()
-	if len(mrs) == 0 {
-		return nil
-	}
 
-	label := mrs[0].GetObjectKind().GroupVersionKind().String()
+	label := mrList.GetObjectKind().GroupVersionKind().String()
```

Comment:
Does this return the same GVK it used to? I think Lists usually have a different kind, e.g. `InstanceList` instead of `Instance`.

Reply:
It was returning an empty GVK, that's why I couldn't notice when I was just checking the number of MRs. I've passed the GVK info from the provider and used it to record metrics.